services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/service_tracker/_models.py

```diff
@@ -66,13 +66,30 @@ class TrackedServiceModel:  # pylint:disable=too-many-instance-attributes
         }
     )
 
-    current_state: SchedulerServiceState = field(
+    _current_state: SchedulerServiceState = field(
         default=SchedulerServiceState.UNKNOWN,
         metadata={
             "description": "to set after parsing the incoming state via the API calls"
         },
     )
 
+    @property
```

Comment:
Q: Why aren’t Pydantic models being used here, especially given their recent performance improvements? I’d be interested in seeing a use case where dataclasses provide a clear advantage.

Reply:
models were replaced with pydantic